libid/complex.cpp

```diff
@@ -0,0 +1,536 @@
+    }
+
+/**************************************************************************
+	cos(x+iy)  = cos(x)cosh(y) - isin(x)sinh(y)
+***************************************************************************/
+
+Complex	Complex::CCos()
+    {
+    Complex a;
+
+    a.x = cos(x) * cosh(y);
+    a.y = -sin(x) * sinh(y);
+    return a;
+    }
+
+/**************************************************************************
+	tan(x+iy)  = (tan(x) -i tanh(-y)) / (1 +i tanh(-y) tan(x))
+***************************************************************************/
+
+Complex	Complex::CTan(void)
+    {
+    Complex a;
+    double  denom;
+
+    x *= 2;
+    y *= 2;
+    denom = cos(x) + cosh(y);
+    if (fabs(denom) <= DBL_MIN) 
+	a = DBL_MAX;
+    else
+	{
+	a.x = sin(x) / denom;
+	a.y = sinh(y) / denom;
+	}
+    return a;
+    }
+
+/**************************************************************************
+	sinh(x+iy)  = sinh(x)cos(y) + icosh(x)sin(y)
+***************************************************************************/
+
+Complex	Complex::CSinh(void)
+    {
+    Complex a;
+
+    a.x = sinh(x) * cos(y);
+    a.y = cosh(x) * sin(y);
+    return a;
+    }
+
+/**************************************************************************
+	cosh(x+iy)  = cosh(x)cos(y) - isinh(x)sin(y)
+***************************************************************************/
+
+Complex	Complex::CCosh(void)
+    {
+    Complex a;
+
+    a.x = cosh(x) * cos(y);
+    a.y = -sinh(x) * sin(y);
+    return a;
+    }
+
+/**************************************************************************
+	tanh(x+iy)  = (tanh(x) + i tan(y)) / (1 +i tanh(x) tan(y))
+***************************************************************************/
+
+Complex	Complex::CTanh(void)
+    {
+    Complex a;
+    double  denom;
+
+    x *= 2;
+    y *= 2;
+
+    denom = cosh(x) + cos(y);
+    if (fabs(denom) <= DBL_MIN) 
+	a = DBL_MAX;
+    else
+	{
+	a.x = sinh(x) / denom;
+	a.y = sin(y) / denom;
+	}
+    return a;
+    }
+
+/**************************************************************************
+	Complex Exponent: e^(x+iy) = (e^x) * cos(y) + i * (e^x) * sin(y)
+***************************************************************************/
+
+Complex	Complex::CExp()
+    {
+    Complex a;
+
+    a.x = exp(x) * cos(y);
+    a.y = exp(x) * sin(y);
+    return a;
+    }
+
+/**************************************************************************
+	sqr(x+iy)  = x*x-y*y + i(x*y)*2
+***************************************************************************/
+
+Complex	Complex::CSqr()
+    {
+    Complex a;
+    double  temp;
+
+    a.x = (x * x) - (y * y);
+    temp = x * y;
+    a.y = temp + temp;
+    return a;
+    }
+
+/**************************************************************************
+	sqrt(x+iy)  =  
+
+DeMoivre’s Theorem states that if n is any positive real number, then
+(a + bi)^n = r^n(cos nθ + i sin nθ).
+
+In particular, if n = ½, we have
+√a + bi = √r [cos θ/2 + i sin θ/2]
```

Comment:
Use only ASCII characters in source files.  No unicode.  This marks the file as binary and prevents diffing and merging.  I explained this in an earlier email when I tried to rebase your branch, I believe.

This has probably 'tainted' this entire pull request because the file will be permanently marked as binary as-is.  Maybe there's a way for a git guru to undo it, but what I saw when attempting to rebase your branch was that it was very difficult to undo this and honestly, I think it will be easier if you just create a new branch, remove the unicode BOM at the front of the file and remove all unicode characters before you add the file.

Because of this situation, I'm going to ask that you try recreating a branch without unicode in the source files and make a new pull request before I continue reviewing this one.  This is a show stopper, unfortunately.